tools/gen_ota_zip.py

```diff
@@ -123,25 +123,9 @@ def gen_diff_ota_sh(patch_path, bin_list, newpartition_list, args, tmp_folder):
 ''' % (ota_progress_list[0])
     fd.write(str)
 
-    if (args.skip_version_check) :
```

Comment:
should we remove all skip_version_check? @JianyuWang0623

Reply:
> should we remove all skip_version_check? @JianyuWang0623

OK, and there are some other works to do. @xiaoxiang781216 

1. Some product / partition may not enable AVB yet
    1. As we plan to update "version_check"(aka upgrade verifiation, anti-downgrade) to avb_verify, all product/partition should enable AVB sign. Will check before pick back to gerrit:trunk.
    2. Considering some partition may not need anti-downgrade, should we add support for avb_verify that if no footer was found then return success directly?
        ```bash
        # Currently: 
        #      Returns success when rollback index of image is equal to / greater than partition.
        # Update:
        #      If no footer was found in image and partition, returns success.
        avb_verify -U /ota/vela_ap.bin /dev/ap /etc/key.avb
        ```
4. Test / Debug needs downgrade maybe
    1. If the rollback index has not been update yet(still default zero), anti-downgrade actually takes no effect yet, it`s OK for auto test.
    2. Else if the rollback index has been updated, the anti-downgrade takes effect, we should unpack the ota.zip and re-sign images/binaries with same rollback index as partition, and then re-generate the ota package.

Reply:
Frist, we should drop all version check based on kvdb because it isn't secure enough. Second, avb base check should be enabled only when the image contains no zero-version number.

Reply:
@xiaoxiang781216 
> should we remove all skip_version_check? @JianyuWang0623

Updated as:
6. ota: Remove the skip_version_check option

> Frist, we should drop all version check based on kvdb because it isn't secure enough.

Removed as:
2. ota: Remove the deprecated ota.version.next and ota.version.current (KV base)
5. ota: Remove the deprecated avb_verify -c upgrade verify (KV base)

> Second, avb base check should be enabled only when the image contains no zero-version number. The final result is that the version check is controlled by avb sign.

Supported by:
3. ota: Update "version_check"(aka upgrade verifiation, anti-downgrade) to AVB (AVB base)